app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextActions.java

```diff
@@ -121,36 +103,33 @@ private boolean onClickImpl(final View view) {
             }
             switch (_action) {
                 case R.string.tmaid_markdown_quote: {
-                    runPrefixReplaceAction(PREFIX_QUOTE, ">$1 ", "");
+                    runRegexReplaceAction(replacePatternGenerator.toogleQuote());
                     return true;
                 }
                 case R.string.tmaid_markdown_h1: {
-                    setHeadingAction(1);
+                    runRegexReplaceAction(replacePatternGenerator.setOrUnsetHeadingWithLevel(1));
                     return true;
                 }
                 case R.string.tmaid_markdown_h2: {
-                    setHeadingAction(2);
+                    runRegexReplaceAction(replacePatternGenerator.setOrUnsetHeadingWithLevel(2));
                     return true;
                 }
                 case R.string.tmaid_markdown_h3: {
-                    setHeadingAction(3);
+                    runRegexReplaceAction(replacePatternGenerator.setOrUnsetHeadingWithLevel(3));
                     return true;
                 }
                 case R.string.tmaid_common_unordered_list_char: {
                     final String listChar = _appSettings.getUnorderedListCharacter();
-                    final String listPrefix = "$1" + listChar + " ";
-                    runPrefixReplaceAction(PREFIX_UNORDERED_LIST, listPrefix, "$1");
+                    runRegexReplaceAction(replacePatternGenerator.replaceWithUnorderedListPrefixOrRemovePrefix(listChar));
                     return true;
                 }
                 case R.string.tmaid_common_checkbox_list: {
                     final String listChar = _appSettings.getUnorderedListCharacter();
-                    final String uncheck = "$1" + listChar + " [ ] ";
-                    final String check = "$1" + listChar + " [x] ";
-                    runPrefixReplaceAction(PREFIX_UNCHECKED_LIST, uncheck, check);
+                    runRegexReplaceAction(replacePatternGenerator.toggleToCheckedOrUncheckedListPrefix(listChar));
                     return true;
                 }
                 case R.string.tmaid_common_ordered_list_number: {
-                    runPrefixReplaceAction(PREFIX_ORDERED_LIST, "$11. ", "$1");
+                    runRegexReplaceAction(replacePatternGenerator.replaceWithOrderedListPrefixOrRemovePrefix());
```

Comment:
At least for this refactoring, I tried out the markdown actions right after the change and didn't notice any behavior change. For regex replacements, I didn't change anything in the other formats, only for markdown.
A good thing about extracting this replacement logic to own classes is that now unit tests can be written for the action behaviors (which I already did for zim while the implementation) to avoid breaking code here in future revisions.

But yeah, it's a good idea to go through the (changed) formats once again at the end to verify that they still work as expected, I will do this.